opencti-platform/opencti-front/src/private/components/data/csvMapper/representations/CsvMapperConditionalEntityMapping.tsx

```diff
@@ -0,0 +1,152 @@
+import React, { FunctionComponent } from 'react';
+import MuiTextField from '@mui/material/TextField';
+import MUIAutocomplete from '@mui/material/Autocomplete';
+import { Field, FieldProps } from 'formik';
+import { CsvMapperColumnBasedFormData, CsvMapperRepresentationFormData } from '@components/data/csvMapper/representations/Representation';
+import Tooltip from '@mui/material/Tooltip';
+import { InformationOutline } from 'mdi-material-ui';
+import { alphabet } from '@components/data/csvMapper/representations/attributes/AttributeUtils';
+import { useFormatter } from '../../../../../components/i18n';
+import TextField from '../../../../../components/TextField';
+import SwitchField from '../../../../../components/fields/SwitchField';
+
+interface CsvMapperConditionalEntityMappingProps
+  extends FieldProps<CsvMapperColumnBasedFormData> {
+  representation: CsvMapperRepresentationFormData;
+  representationName: string;
+}
+const CsvMapperConditionalEntityMapping: FunctionComponent<
+CsvMapperConditionalEntityMappingProps
+> = ({ form, representationName, representation }) => {
+  const { t_i18n } = useFormatter();
+  const columnOptions = alphabet(26);
+  const operatorOptions = [
+    { label: t_i18n('Equal'), value: ('eq') },
+    { label: t_i18n('Not equal'), value: 'not_eq' }];
+  const { setFieldValue } = form;
+
+  const handleColumnSelect = async (column: string | null) => {
+    await setFieldValue(`${representationName}.column_based.column_reference`, column);
+  };
+
+  const handleOperatorSelect = async (operator: { label: string, value: string } | null) => {
+    await setFieldValue(`${representationName}.column_based.operator`, operator?.value);
+  };
+
+  return (
+    <div style={{
+      width: '100%',
+      display: 'grid',
+      gridTemplateColumns: '1.3fr 1fr 1fr 1fr',
+      alignItems: 'center',
+      margin: '30px 0px',
+      gap: '10px',
+    }}
+    >
+      <div style={{
+        display: 'flex',
+        alignItems: 'center',
+      }}
+      >
+        <Field
+          component={SwitchField}
+          type="checkbox"
+          name={`${representationName}.column_based.enabled`}
+          label={t_i18n('Entity dynamic mapping')}
+        />
+        <Tooltip
+          title={t_i18n(
+            'If this option is selected, we will dynamically map the column value that you provide to the entity.',
+          )}
+        >
+          <InformationOutline
+            fontSize="small"
+            color="primary"
+            style={{ cursor: 'default' }}
+          />
+        </Tooltip>
+      </div>
+      <MUIAutocomplete
+        selectOnFocus
+        openOnFocus
+        autoSelect={false}
+        autoHighlight
+        options={columnOptions}
+        disabled={!representation.column_based?.enabled}
```

Comment:
For easy reading, should we create a variable as `representation.column_based` is used many times ?